code/modules/reagents/chemistry/reagents/potion_reagents.dm

```diff
@@ -0,0 +1,129 @@
+/*
+ *Magical elixirs for potion flasks. Identify them via your local chemist or test your luck. Whats the worst that could happen?
+ */
+
+
+/datum/reagent/consumable/potion
+	name = "strange elixir"
+	id = "genericpotion"
+	description = "A strange mixture of various components. There appears to be no real identifier to what its suppose to do."
+	color = "#FFF804" // rgb: 225, 248, 4
+	nutriment_factor = 0
+
+
+//healing potion
+/datum/reagent/consumable/potion/healing
+	id = "healingpotion"
+	description = "A strange mixture of various components. Pleasant fumes drift aimlessly off the mixture, just smelling it makes you feel better."
+	color = "#FF0000" // rgb: 255, 0, 0
+
+/datum/reagent/consumable/potion/healing/reaction_mob(mob/living/M, method=INGEST, reac_volume, show_message = 1)
+	if(iscarbon(M) && M.stat != DEAD)
+		if(method in list(TOUCH, VAPOR, PATCH))
+			if(show_message)
+				M << "<span class='notice'>Your skin tingles as you feel your ailments seem to almost fade away.</span>"
+				M.adjustOxyLoss(-3 * reac_volume)
+				M.adjustBruteLoss(-3 * reac_volume)
+				M.adjustFireLoss(-3 * reac_volume)
+		else
+			if(show_message)
+				M << "<span class='notice'>A rejuvenating rush of energy travels throughout your body, you've never felt better!</span>"
+				M.adjustOxyLoss(-5 * reac_volume)
+				M.adjustBruteLoss(-5 * reac_volume)
+				M.adjustFireLoss(-5 * reac_volume)
+				M.dizziness = 0
+				M.drowsyness = 0
+				M.stuttering = 0
+				M.confused = 0
+				M.radiation = 0
+	..()
+
+
+//poison potion
+/datum/reagent/consumable/potion/poison
+	id = "poisonpotion"
+	description = "A strange mixture of various components. Foul fumes drift off the horrible mixture as it occassionally bubbles. Why would anyone make this?"
+	color = "#00FF00" // rgb: 0, 255, 0
+
+/datum/reagent/consumable/potion/poison/reaction_mob(mob/living/M, method=INGEST, reac_volume, show_message = 1)
+	if(iscarbon(M) && M.stat != DEAD)
+		if(method in list(TOUCH, VAPOR, PATCH))
+			if(show_message)
+				M << "<span class='userdanger'>A horrible pain washes over you, it feels like something is eating away at your skin!</span>"
+				M.adjustBruteLoss(25)
```

Comment:
This is weird. Wouldn't you just hit the target with 1u, wait for it to decay, then hit them again?